### ISSUE TITLE
[Bug]: Custom fields index table is has wrong html structure

### ISSUE THREAD

**Issue by DonKoko** [Shelf-nu/shelf.nu#471]
### What happened?

Steps to reproduce:

1. Go to Settings > Custom fields
2. Inspect the table elements
3. Notice how the link is not covering the whole row
<img width="1800" alt="Screenshot 2023-10-20 at 13 46 22" src="https://github.com/Shelf-nu/shelf.nu/assets/7840007/8bf68b25-ffdb-46c9-8122-dcae1903f7a6">

We actually have built in functionality to our table components to handle navigating to prevent this issue specifically.

### What is the expected behaviour?

You should be able to navigate to a custom fields page by clicking anywhere within its row in the index table.

### Version

Hosted (app.shelf.nu)

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

**Comment by hunar1312** [comment]
hey @DonKoko, as discussed at the time of development, if converted whole row as a navigating element then the actions dropdown won't be clickable. Although I see the whole name column isn't a link, looks like the contributor who worked on adding differet types of custom fields missed this. I'll update that.

**Comment by DonKoko** [comment]
Yeah i also missed it on the review. lets at least make the name column a link.
